ibis/backends/snowflake/tests/test_udf.py

```diff
@@ -150,3 +154,108 @@ def cases(value, mapping):
+    for root, dirs, files in os.walk(pkgpath):
+        for path in dirs + files:
+            os.utime(os.path.join(root, path), None)
+
+
+def add_packages(d, session):
+    import parsy
+    import pyarrow_hotfix
+    import rich
+    import sqlglot
+
+    import ibis
+
+    for module in (rich, parsy, sqlglot, pyarrow_hotfix):
+        pkgname = module.__name__
+        pkgpath = os.path.join(d, pkgname)
+        shutil.copytree(os.path.dirname(module.__file__), pkgpath)
+        touch_package(pkgpath)
+        session.add_import(pkgname, import_path=pkgname)
+
+    # no need to touch the package because we're using the local version
+    shutil.copytree(os.path.dirname(ibis.__file__), "ibis")
+    session.add_import("ibis", import_path="ibis")
+
+
+@pytest.fixture
+def snowpark_session():
+    if not os.environ.get("SNOWFLAKE_SNOWPARK"):
+        pytest.skip("SNOWFLAKE_SNOWPARK is not set")
+    else:
+        sp = pytest.importorskip("snowflake.snowpark")
+
+        if connection_name := os.environ.get("SNOWFLAKE_DEFAULT_CONNECTION_NAME"):
+            builder = sp.Session.builder.config("connection_name", connection_name)
+        else:
+            builder = sp.Session.builder.configs(
+                {
+                    "user": os.environ["SNOWFLAKE_USER"],
+                    "account": os.environ["SNOWFLAKE_ACCOUNT"],
+                    "password": os.environ["SNOWFLAKE_PASSWORD"],
+                    "warehouse": os.environ["SNOWFLAKE_WAREHOUSE"],
+                    "database": os.environ["SNOWFLAKE_DATABASE"],
+                    "schema": os.environ["SNOWFLAKE_SCHEMA"],
+                }
+            )
+
+        session = builder.create()
+        session.custom_package_usage_config["enabled"] = True
+
+        pwd = os.getcwd()
+
+        with tempfile.TemporaryDirectory() as d:
+            os.chdir(d)
+
+            try:
+                add_packages(d, session)
+                yield session
+            finally:
+                os.chdir(pwd)
+
+
+def test_ibis_inside_snowpark(snowpark_session):
+    sp = pytest.importorskip("snowflake.snowpark")
+
+    def run(session):
+        import ibis
+
+        con = ibis.backends.snowflake.Backend.from_snowpark(session)
+        t = con.tables.functional_alltypes
+        expr = t.group_by("string_col").agg(n=t.count()).order_by("string_col")
+        return session.sql(ibis.to_sql(expr))
+
+    local_result = run(snowpark_session).to_pandas()
+
+    assert local_result.columns.tolist() == ["string_col", "n"]
+
+    sproc_name = "the_ibis"
+    snowpark_session.sproc.register(
+        run,
+        name=sproc_name,
+        imports=["parsy", "rich", "sqlglot", "pyarrow_hotfix", "ibis"],
+        return_type=sp.types.StructType(),
+        packages=[
+            "snowflake-snowpark-python",
+            "toolz",
+            "atpublic",
+            "bidict",
+            "pyarrow",
+            "pandas",
+            "numpy",
+        ],
+        replace=True,
```

Comment:
I was able to get this test passing registering the stored procedure with owner's rights. I did recieve an internal error when trying to register as caller's rights, but that allowed for the session parameters to be adjusted as I suspected but caused an errored out the stored procedure itself.  

Could you try adding the `execute_as="caller",` arg on the stored procedure registration to confirm that this error is not caused by an account setting on my side?

Reply:
I get this error when I pass `execute_as="caller"`:

```
E       snowflake.connector.errors.ProgrammingError: 000603 (XX000): SQL execution internal error:
E       Processing aborted due to error 300010:3474929812; incident 3000484.
```